xhttp/xhttp.go

```diff
@@ -13,6 +13,9 @@ type (
 	// instead of a response body.
 	Response[T any] struct {
 		*http.Response
+		// RawObj is the raw response from where Obj was parsed (useful mostly for debugging).
+		RawObj []byte
```

Comment:
Not sure about this name, open to suggestions (didn't want to shadow the Response.Body).